internal/health_checks/network_check.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package health_checks
+
+import (
+    "io"
+    "fmt"
+    "net/http"
+    "github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+)
+
+var (
+    // API urls
+    loggingAPIUrl = "https://logging.googleapis.com/$discovery/rest"
+    monitoringAPIUrl = "https://monitoring.googleapis.com/$discovery/rest"
+)
+
+func runGetHTTPRequest(url string) (string, string, error) {
+    resp, err := http.Get(url)
+    if err != nil {
+        return "", "", err
+    }
+    defer resp.Body.Close()
+
+    status := resp.Status
+    b, err := io.ReadAll(resp.Body)
+    if err != nil {
+        return "", "", err
+    }
+
+    return status, string(b), nil
+}
+
+type NetworkCheck struct{
+    HealthCheck
+}
+
+func (c NetworkCheck) RunCheck(uc *confgenerator.UnifiedConfig) error {
+
+    // Request to logging API
+    status, _, err := runGetHTTPRequest(loggingAPIUrl)
+    c.LogMessage("==>" + status)
+    if err != nil {
+        fmt.Println(err)
+        return err
+    }
```

Comment:
According to [the docs for http.Get](https://pkg.go.dev/net/http#Get), "An error is returned if there were too many redirects or if there was an HTTP protocol error. "

In particular, if the network we are on has issues such as improper proxies, I can see an infinite redirect situation happening. 

Should this situation be handled as an error or a test failure?

Reply:
Yes, I think an overall thing to address (see the last commits I added) is how differentiate clearly between `check failure` and `error`.
- How to report an error ?
- Should we handle the errors ?
- How to differentiate this in very detailed cases, e.g. `http.Get` ?